go-controller/pkg/ovn/gateway_init.go

```diff
@@ -537,18 +538,25 @@ func addPolicyBasedRoutes(nodeName, mgmtPortIP string, hostIfAddr *net.IPNet) er
 
 	// embed nodeName as comment so that it is easier to delete these rules later on.
 	// logical router policy doesn't support external_ids to stash metadata
-	matchStr := fmt.Sprintf(`inport == "%s%s" && %s.dst == %s /* %s */`,
-		types.RouterToSwitchPrefix, nodeName, l3Prefix, hostIfAddr.IP.String(), nodeName)
-	if err := syncPolicyBasedRoutes(nodeName, mgmtPortIP, matchStr, types.NodeSubnetPolicyPriority, mgmtPortIP, "match,nexthops"); err != nil {
+	hostIPs := make([]string, len(otherHostAddrs))
+	copy(hostIPs, otherHostAddrs)
```

Comment:
Question: why do we need to copy it by the way?

Reply:
so that I dont mangle the incoming slice. I could just use a set here instead. what do you prefer?

Reply:
I've gone ahead and changed this.